en/docs/get-started/key-concepts.md

```diff
@@ -1,128 +1,147 @@
+#### Message Stores and Processors
 
-### Message Stores and Processors
+A Message Store is used to temporarily store messages before they are delivered to their destination. This approach is useful for several scenarios.
 
-A **Message Store** is used by a [mediation sequence](#mediation-sequences) to temporarily store messages before they are delivered to their destination. This approach is useful for several scenarios.
+1. Serving traffic to back-end services that can only accept messages at a given rate, whereas incoming traffic arrives at different rates. This use case is called request rate matching.
+2. If the back-end service is not available at a particular moment, the message can be kept safely inside the message store until the back-end service becomes available. This use case is called guaranteed delivery. 
 
-1. Serving traffic to back-end services that can only accept messages at a given rate, whereas incoming traffic arrives at different rates. This use case is called **request rate matching**.
-2. If the back-end service is not available at a particular moment, the message can be kept safely inside the message store until the back-end service becomes available. This use case is called **Guaranteed delivery**. 
+The task of the Message Processor is to pick the messages stored in the message store and deliver them to the destination.
 
-The task of the **Message Processor** is to pick the messages stored in the Message Store and deliver them to the destination. 
+See the [Message Stores and Processors]({{base_path}}/reference/synapse-properties/about-message-stores-processors) documentation for more information.
 
-### Templates
+#### Templates
 
-A large number of configuration files in the form of [sequences](#mediation-sequences) and [endpoints](#endpoints), and transformations can be required to satisfy all the mediation requirements of your system. 
-To keep your configurations manageable, it's crucial to avoid spreading configuration files across various locations and to eliminate redundant duplications. Templates help minimize this redundancy by creating prototypes that users can use and reuse when needed. WSO2 Micro Integrator can template [sequences](#mediation-sequences) and [endpoints](#endpoints).
+A Template is a reusable configuration that helps streamline and simplify the creation and management of [sequences](#sequences) and [endpoints](#endpoints). Templates allow developers to abstract common logic and reuse it across multiple services, reducing redundancy and ensuring consistency in configurations.
+
+See the [Templates]({{base_path}}/reference/synapse-properties/template-properties) documentation for more information.
 
 ---
 
-## Message exit points
+### Message exit points
+
+#### Endpoints
 
-### Endpoints
+An Endpoint defines an external destination for a message. An endpoint could represent an HTTP URL, a mailbox, a JMS queue, a TCP socket, etc., along with the settings needed for the connection. 
```

Comment:
```suggestion
An endpoint defines an external destination for a message. It could represent an HTTP URL, a mailbox, a JMS queue, a TCP socket, and other connection types, along with the settings needed for the connection.
```